Lombiq.Tests.UI/SecurityScanning/SecurityScanningConfiguration.cs

```diff
@@ -30,7 +32,23 @@ public class SecurityScanningConfiguration
     /// </summary>
     public Action<UITestContext, SarifLog> AssertSecurityScanResult { get; set; } = AssertSecurityScanHasNoAlerts;
 
-    public static readonly Action<UITestContext, SarifLog> AssertSecurityScanHasNoAlerts =
-        (_, sarifLog) => sarifLog.Runs[0].Results.ShouldNotContain(result =>
-            result.Kind == ResultKind.Fail && result.Level != FailureLevel.None && result.Level != FailureLevel.Note);
+    public static readonly Action<UITestContext, SarifLog> AssertSecurityScanHasNoAlerts = (_, sarifLog) =>
+    {
+        var errors = sarifLog
+            .Runs[0]
+            .Results
+            .Where(result =>
+                result.Kind == ResultKind.Fail &&
+                result.Level != FailureLevel.None &&
+                result.Level != FailureLevel.Note)
+            .Select(result => new
+            {
+                Kind = result.Kind.ToString(),
+                Level = result.Level.ToString(),
+                Details = result,
+            })
+            .ToList();
+
+        errors.ShouldBeEmpty(JsonConvert.SerializeObject(errors));
```

Comment:
If `errors` is not empty, this serializes the affected items into the custom message of the `ShouldAssertException`. This is amazing for debugging, because the exception delivers you the exact problem instead of the whole collection of junk that gets truncated in the output anyway. The overhead is minima, during normal operation, it only creates an empty list and then serializes it. That's pretty cheap and well worth the infinite improvement in terms of developer experience.

Reply:
I see, that's useful, just please comment it because it's not apparent.

Reply:
I do you one better! I turned it into an extension method for future reuse in other tests.

Reply:
Ah, I wanted to ask that, great,